### ISSUE TITLE
test failure on develop branch

### ISSUE THREAD

**Issue by amueller** [openml/openml-python#331]
```
$ nosetests -sv tests/test_tasks/test_task_functions.py -m test_get_task_with_cache 
```
```
======================================================================
ERROR: test_get_task_with_cache (tests.test_tasks.test_task_functions.TestTask)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/andy/checkout/openml-python/tests/test_tasks/test_task_functions.py", line 108, in test_get_task_with_cache
    task = openml.tasks.get_task(1)
  File "/home/andy/checkout/openml-python/openml/tasks/functions.py", line 247, in get_task
    task.download_split()
  File "/home/andy/checkout/openml-python/openml/tasks/task.py", line 83, in download_split
    split = OpenMLSplit._from_arff_file(cached_split_file)
  File "/home/andy/checkout/openml-python/openml/tasks/split.py", line 120, in _from_arff_file
    return cls(name, '', repetitions)
  File "/home/andy/checkout/openml-python/openml/tasks/split.py", line 27, in __init__
    self.split[repetition][fold][sample] = split[repetition][fold][sample]
TypeError: only integer scalar arrays can be converted to a scalar index
-------------------- >> begin captured logging << --------------------
fasteners.process_lock: Level 5: Acquired file lock `/home/andy/checkout/openml-python/tests/files/locks/datasets.functions.get_dataset:1` after waiting 0.000s [1 attempts were required]
fasteners.process_lock: Level 5: Unlocked and closed file lock open on `/home/andy/checkout/openml-python/tests/files/locks/datasets.functions.get_dataset:1`
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 1 test in 144.598s

FAILED (errors=1)
```

**Comment by amueller** [comment]
So I needed to remove ``openml-python/tests/files/tasks/1/datasplits.pkl`` which was not shown by git... so our tests should all clean up after themselves and we shouldn't add this stuff to ``.gitignore`` because that leads to this kind of issues.

**Comment by amueller** [comment]
What contributes to the problem is that some files in ``tests/files`` are actually part of the test-suite and some are generated, so it's unclear which to delete for a fresh run. But if we don't keep any at all, that'll solve the problem. Having files stored there also means that the outcome can depend on the order of tests.

**Comment by mfeurer** [comment]
Resolved via #336.
